server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -909,7 +909,11 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
             } else if (operation == FielddataOperation.SEARCH) {
                 if (fielddata == false) {
                     throw new IllegalArgumentException(
-                        "Text fields are not optimised for operations that require per-document "
+                        "Field data is disabled on ["
```

Comment:
super minor nit: could you remove the whitespace between Field and data? I've seen that that's how we refer to it in our docs and I would prefer to keep that consistent.

Reply:
```suggestion
                        "Fielddata is disabled on ["
```